### PR TITLE
Add pkgconfig file

### PR DESCRIPTION
**Pull request by sl1pkn07**
generate and install pkgconfig file (https://www.freedesktop.org/wiki/Software/pkg-config/)

related: https://github.com/NVIDIA/nccl/issues/186

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -77,6 +81,16 @@ $(LIBDIR)/$(STATICLIBTARGET): $(LIBOBJ) $(DEVICELIB)
 	ar cr $@ $(LIBOBJ) $(TMP)/*.o
 	rm -Rf $(TMP)
 
+$(PKGDIR)/nccl.pc : nccl.pc.in
+	@mkdir -p $(PKGDIR)
+	printf "Generating %-35s > %s\n" $< $@
+	printf $(PREFIX)
```
Comment (sjeaugey) [inline src/Makefile]: Not sure if that printf was supposed to stay

File: src/Makefile [hunk 2]
```diff
@@ -77,6 +81,16 @@ $(LIBDIR)/$(STATICLIBTARGET): $(LIBOBJ) $(DEVICELIB)
...
+	sed -e 's|$${nccl:Prefix}|\$(PREFIX)|g' \
```
Comment (sjeaugey) [inline src/Makefile]: Is there a reason for this sed expr being different from the others below ? Note I don't have a preference, just looking for consistency.

Comment (sl1pkn07) [reply src/Makefile]: because i test with other form and sed fails, i think because the '$(PREFIX)', the start character is '/'

Comment (sl1pkn07) [reply src/Makefile]: the generated pc file is:

~~~
prefix=/usr/local
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: nccl
Description: Optimized primitives for collective multi-GPU communication
Version: 2.4.2
Libs: -L${libdir} -lnccl
Cflags: -I${includedir}
~~~
(makefile launched with: make CUDA_HOME='/opt/cuda' -j40 V=1)

File: src/Makefile (hunk 1 above)
Comment (sl1pkn07) [reply src/Makefile]: oh, yes, is a typo

fixing

**Pull request closed without merging**